barretenberg/cpp/pil/avm/mem.pil

```diff
@@ -176,10 +176,11 @@ namespace mem(256);
     (1 - lastAccess) * (1 - rw') * (tag' - tag) = 0;
 
     // Constrain that the first load from a given address has value 0. (Consistency of memory initialization.)
-    // We do not constrain that the tag == 0 as the 0 value is compatible with any memory type.
     // As we enforce lastAccess = 1 on the first row, the following condition applies also for the first memory entry:
     #[MEM_ZERO_INIT]
     lastAccess * (1 - rw') * val' = 0;
+    // Constrain that reading an uninitialized cell creates tag error unless it is expected to be of type FF.
+    lastAccess * (1 - rw') * ((r_in_tag - constants.MEM_TAG_FF) * (1 - one_min_inv) - tag_err) = 0;
```

Comment:
I think in this precise use case, you should not involve one_min_inv value or tag_err.
Rather just enforce that tag == FF. Then, #[MEM_IN_TAG_CONSISTENCY_1] and #[MEM_IN_TAG_CONSISTENCY_2] should do the job of constraining tag_err == 1 if r_in_tag is not FF.

Reply:
#[MEM_TAG_INIT]
lastAccess * (1 - rw') * (tag' - constants.MEM_TAG_FF) = 0;